src/server/storage/storage.js

```diff
@@ -357,35 +361,53 @@ Storage.prototype.loadPaths = function (data, callback) {
 
     this.mongo.openProject(data.projectId)
         .then(function (dbProject) {
-            var loadedObjects = {};
-
-            //self.logger.error('paths:', data.paths);
+            var loadedObjects = {},
+                throttleDeferred = Q.defer(),
```

Comment:
what is saved here by having this 'throttle' function? shouldn't we use a library throttle (and for the loadObjects instead of the loadPath(s))?